packages/flutter/lib/src/widgets/autocomplete.dart

```diff
@@ -58,6 +58,38 @@ typedef AutocompleteOptionsViewBuilder<T extends Object> = Widget Function(
   Iterable<T> options,
 );
 
+/// The type of the [RawAutocomplete] callback which returns a [Widget] that
+/// can be displayed while [optionsBuilder] is being resolved.
+///
+/// The returned widget from this callback will be wrapped in an
+/// [AutocompleteHighlightedOption] inherited widget. This will allow
+/// this callback to determine which option is currently highlighted for
+/// keyboard navigation.
+///
+/// See also:
+///
+///   * [RawAutocomplete.optionsViewBuilder], which is of this type.
```

Comment:
This should be loadingStateBuilder, not optionsViewBuilder. Similarly for emptyStateBuilder below.